src/qibo/models/encodings.py

```diff
@@ -1483,3 +1483,73 @@ def _get_phase_gate_correction_sparse(
         gate = _get_phase_gate_correction(last_string, phis[-1])
 
     return gate
+
+
+def dicke_state_encoder(nqubits: int, k: int, **kwargs):
+    """Create a circuit that prepares the Dicke state |D_k^n⟩.
+
+    The Dicke state |D_k^n⟩ is the equal superposition of all n-qubit states
+    with Hamming weight k (exactly k ones). The circuit prepares the state
+    deterministically with O(kn) gates and O(n) depth, following the construction
+    from arXiv:1904.07358.
+
+    Args:
+        nqubits (int): Total number of qubits (n).
+        k (int): Hamming weight of the Dicke state (k ≤ n).
+        kwargs: Additional arguments for Circuit initialization.
+
+    Returns:
+        Circuit: Quantum circuit that prepares |D_k^n⟩ when applied to |0⟩^⊗n.
+    """
```

Comment:
It would be good if the docstring followed the pattern of the other docstrings in the module. For instance, see https://github.com/qiboteam/qibo/blob/5fe998de11072407e12275819803e6fed49caf26/src/qibo/models/encodings.py#L483, especially in terms of how the math is typed.

```suggestion
def dicke_state(nqubits: int, weight: int, backend=None, **kwargs):
    """Create a circuit that prepares the Dicke state |D_k^n⟩.

    The Dicke state |D_k^n⟩ is the equal superposition of all n-qubit states
    with Hamming weight k (exactly k ones). The circuit prepares the state
    deterministically with O(kn) gates and O(n) depth, following the construction
    from arXiv:1904.07358.

    Args:
        nqubits (int): number of qubits :math:`n`.
        weight (int): Hamming weight :math:`k` of the Dicke state.
        kwargs: Additional arguments for Circuit initialization.

    Returns:
        :class:`qibo.models.circuit.Circuit` : Quantum circuit that prepares |D_k^n⟩ when applied to |0⟩^⊗n.
    """
```